### ISSUE TITLE
usage question: does geminabox solve for serving different architectures?

### ISSUE THREAD

**Issue by josh-m-sharpe** [geminabox/geminabox#563]
TL;DR
Can I push a compiled version of a gem to geminabox for different architectures so the clients that are bundling a gemfile don't need to compile native extensions?

------
Situation:  [grpc](https://rubygems.org/gems/grpc) (the nemesis of anyone who uses a google client) takes for freaking ever to compile... and then it produces like a full GB of whatever in vendor/bundle.

We currently use it on intel and arm machines.  And every time they push a new version it takes for freaking ever, again, across multiple machines.  It's legit a pile of terrible google-driven ruby.

I'm wondering if I can compile them once, push the resulting artifacts to geminabox and then other machines that need it can just pull from geminabox and skip the compilation step.  Is this a thing I can do with geminabox?

This obviously doesn't just apply to grpc but any gem that has native extensions.  Generally it would speed up bundling in our multi-architecture environment.

Thanks!

**Comment by ahorek** [comment]
you have to generate precompiled gems for each platform, see https://github.com/rake-compiler/rake-compiler and push them all, but the workflow is no different than releasing a gem to standard rubygems.org
